native-src/android/app/src/main/java/com/telerik/localnotifications/NotificationPublisher.java

```diff
@@ -1,53 +1,40 @@
 package com.telerik.localnotifications;
 
-import android.annotation.SuppressLint;
-import android.app.Notification;
 import android.app.NotificationManager;
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
-import android.content.SharedPreferences;
 import android.util.Log;
 
-import org.json.JSONException;
 import org.json.JSONObject;
 
-import static com.telerik.localnotifications.NotificationRestoreReceiver.SHARED_PREFERENCES_KEY;
-import static com.telerik.localnotifications.NotificationRestoreReceiver.TAG;
-
 public class NotificationPublisher extends BroadcastReceiver {
 
-  public static String NOTIFICATION_ID = "notification-id";
-  public static String NOTIFICATION = "notification";
-  public static String SOUND = "sound";
-  public static String PUSH_BUNDLE = "pushBundle";
+  private static final String TAG = "NotificationPublisher";
+
+  public static final String NOTIFICATION_ID = "NOTIFICATION_ID";
 
   public void onReceive(Context context, Intent intent) {
-    final Notification notification = intent.getParcelableExtra(NOTIFICATION);
-    if (notification != null) {
-      //notification.sound = Uri.parse("android.resource://" + context.getPackageName() + "/raw/" + notification.sound);
-      final int id = intent.getIntExtra(NOTIFICATION_ID, 0);
-      ((NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE)).notify(id, notification);
-      removeNotificationIfNotRepeating("" + id, context);
+    final int id = intent.getIntExtra(NOTIFICATION_ID, 0);
+    final JSONObject opts = Store.get(context, id);
+
+    if (opts == null) {
+      Log.e(TAG, "Notification could not be restored, options are null");
+
+      return;
```

Comment:
Not needed?

Reply:
The `return`? Why? It's a guard clause, I don't want to execute what comes next if `opts` is `null`. There could also be an `if (opts != null)` wrapping all the block, but I just prefer this style.

Reply:
It's OK. It's just the diff format and the lack of format didn't allow me to properly visualize the block after this. :p